nntp/newsrc.c

```diff
@@ -92,7 +92,7 @@ static struct NntpMboxData *mdata_find(struct NntpAccountData *adata, const char
   if (adata->groups_num >= adata->groups_max)
   {
     adata->groups_max *= 2;
-    mutt_mem_realloc(&adata->groups_list, adata->groups_max * sizeof(mdata));
+    mutt_mem_reallocarray(&adata->groups_list, adata->groups_max, sizeof(mdata));
   }
```

Comment:
Also fixed here.  BTW, again, the type in the macro helped realize that something was wrong in this call.  Without the type in the macro, it would have automagically deduced `void *`, which works (because all pointers were made equal by $deity), but isn't a great thing.